sycl/doc/design/OffloadDesign.md

```diff
@@ -0,0 +1,396 @@
+
+After the device binaries are linked together, two additional steps are
+performed to prepare the device binary for consumption by an offline compilation
+tool for AOT or to be wrapped for JIT processing.
+
+The `sycl-post-link` tool is used after the device link is performed,
+applying any changes such as optimizations and code splitting before passing
+off to the `llvm-spirv` tool, which translates the LLVM-IR to SPIR-V.
+
+At this point, the SPIR-V binary is sent to the Ahead of Time compilation
+step to produce the final device binary or is wrapped before being linked into
+the final executable.
+
+### Ahead Of Time Compilation
+
+The updated offloading model will integrate the Ahead of Time (AOT) compilation
+behaviors into the clang-linker-wrapper.  The actions will typically take place
+after the device link and post link steps.
+
+Regardless of the AOT target, the flow is similar, only modifying the offline
+compiler that is used to create the target device image.  It is expected that
+the offline compiler will also use unique command lines specific to the tool
+to create the image.
+
+To support the needed option passing triggered by use of the
+`-Xsycl-target-backend` option and implied options based on the optional
+device behaviors for AOT compilations for GPU new command line interfaces
+are needed to pass along this information.
+
+| Target | Triple        | Offline Tool   | Option for Additional Args |
+|--------|---------------|----------------|----------------------------|
+| CPU    | spir64_x86_64 | opencl-aot     | `--cpu-tool-arg=<arg>`     |
+| GPU    | spir64_gen    | ocloc          | `--gen-tool-arg=<arg>`     |
+| FPGA   | spir64_fpga   | aoc/opencl-aot | `--fpga-tool-arg=<arg>`    |
+
+*Table: Ahead of Time Info*
+
+To complete the support needed for the various targets using the
+`clang-linker-wrapper` as the main interface, a few additional options will
+be needed to communicate from the driver to the tool.  Further details of usage
+are given further below.
+
+| Option Name                  | Purpose                                      |
+|------------------------------|----------------------------------------------|
+| `--fpga-link-type=<arg>`     | Tells the link step to perform 'early' or 'image' processing to create archives for FPGA |
+| `--parallel-link-sycl=<arg>` | Provide the number of parallel jobs that will be used when processing with `llvm-foreach` |
+| `--no-sycl-device-lib=<arg>` | Provide the list of device libraries to restrict from linking during device link |
+
+*Table: Additional Options for clang-linker-wrapper*
+
+The `clang-linker-wrapper` provides an existing option named `-wrapper-jobs`
+that may be useful for our usage instead of creating a new option specific
+to `llvm-foreach` processing.
+
+#### spir64_gen support
+
+Compilation behaviors involving AOT for GPU involve an additional call to
+the OpenCL Offline compiler (OCLOC).  This call occurs after the post-link
```

Comment:
Are we missing the LLVM to SPIR-V translation step here and in sections below? Please check.

Thanks